subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestClassProcessor.java

```diff
@@ -95,9 +101,40 @@ public void execute(@Nonnull String testClassName) {
 
         private void processAllTestClasses() {
             Launcher launcher = LauncherFactory.create();
-            launcher.registerTestExecutionListeners(new JUnitPlatformTestExecutionListener(resultProcessor, clock, idGenerator));
-            launcher.execute(createLauncherDiscoveryRequest(testClasses));
+            TestExecutionListener listener = new JUnitPlatformTestExecutionListener(resultProcessor, clock, idGenerator);
+            LauncherDiscoveryRequest discoveryRequest = createLauncherDiscoveryRequest(testClasses);
+            launcher.registerTestExecutionListeners(listener);
+
+            if (spec.isDryRun()) {
+                TestPlan testPlan = launcher.discover(discoveryRequest);
+                executeDryRun(testPlan, listener);
+            } else {
+                launcher.execute(createLauncherDiscoveryRequest(testClasses));
+            }
+        }
+    }
+
+    private void executeDryRun(TestPlan testPlan, TestExecutionListener listener) {
+        List<TestIdentifier> testIdentifiers = new LinkedList<>();
```

Comment:
Probably it have to be a graph of TestIdentifier, not linkedList, since testPlan is actually a graph